modules/display.c

```diff
@@ -3775,6 +3777,74 @@ static void mdy_blanking_start_adaptive_dimming(void)
 
 // AUTOMATIC BLANKING STATE MACHINE
 
+/** Check if blanking inhibit mode denies turning display off
+ *
+ * @return true if automatic blanking should not happen, false otherwise
+ */
+static bool mdy_blanking_inhibit_off_p(void)
+{
+    bool inhibit = false;
```

Comment:
As long as the logic doesn't become any more complicated, this might be easier as:

```
return (system_state != MCE_STATE_ACTDEAD &&
        ((mdy_blanking_inhibit_mode == INHIBIT_STAY_DIM) ||
         (mdy_blanking_inhibit_mode == INHIBIT_STAY_DIM_WITH_CHARGER && charger_state == CHARGER_STATE_ON)));
```

(i might have gotten some conditions wrong, but something along those lines)